### PR TITLE
[FE-190] Add variant report task to VariantAnntoation.wdl

### PR DESCRIPTION
**Pull request by samanthahv**
Adding a task to the VariantAnnotation.wdl that runs a python script that does the following:
- takes the positions json file created from the previous task as input, as well as the sample ID
- loops through the positions json and selects transcripts and variants to include based on the rules in [this spec](https://docs.google.com/document/d/1C0dBQIyuvU15CO3rXcIkDTKAtzwSxaWpVH53PS2ZtsE/edit)
- creates a pdf report and tsv of selected variants

The wdl was tested in the following ToA workspace: https://app.terra.bio/#workspaces/TerraWorkflows/fe-nirvana-json-wdl-dev-az/workflows-app/submission-history/f892503a-7f4e-422c-9161-377a8efb6e7e
image

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# dev
```
Comment (nikellepetrillo) [inline .gitignore:1]: i think we can remove this file from the pr!

File: pipelines/cohort_screening/src/MANE.GRCh38.v1.3.ensembl_genomic.gtf.gz
Comment (nikellepetrillo) [inline pipelines/cohort_screening/src/MANE.GRCh38.v1.3.ensembl_genomic.gtf.gz:1]: can probably remove this file as well! ( i think? )

File: .gitignore (hunk 1 above)
Comment (bahill) [reply .gitignore:1]: oh - those are local development files I don't want added to GitHub, that Intellij and I generated, and we wouldn't want to be committed from anyone else's local dev env either.

File: pipelines/cohort_screening/src/MANE.GRCh38.v1.3.ensembl_genomic.gtf.gz
Comment (bahill) [reply pipelines/cohort_screening/src/MANE.GRCh38.v1.3.ensembl_genomic.gtf.gz:1]: Actually no.[ That's required for the filtering,](https://github.com/broadinstitute/TerraWorkflows/blob/9f9316b09d74ac8b79a99bbd14db10843bfbec2b/pipelines/cohort_screening/src/variants_report.py#L13) and we don't want to have to go and get it each time. This way we know it's always the file we think it is.

File: .gitignore (hunk 1 above)
Comment (bahill) [reply .gitignore:1]: I did change the description to be more clear

File: pipelines/cohort_screening/src/variants_report.py [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+    # do the filtering
+    filtered_positions = filter_transcripts(positions)
+
+    # output the filtered positions to a json file if specified
+    if args.output_json is False:
+        pass
+    else:
+        with open("../local_testing/filtered_positions_final_test.json", 'w') as outfile:
+            json.dump(filtered_positions, outfile)
+            # json.dump(filtered_positions, outfile, indent=4)  # add indent for pretty print, remove for py reading
```
Comment (kilonzi) [inline pipelines/cohort_screening/src/variants_report.py:415]:
```suggestion
```

File: pipelines/cohort_screening/src/variants_report.py [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+    positions = json.load(positions_json)
+
+    # do the filtering
+    filtered_positions = filter_transcripts(positions)
+
+    # output the filtered positions to a json file if specified
+    if args.output_json is False:
+        pass
+    else:
+        with open("../local_testing/filtered_positions_final_test.json", 'w') as outfile:
```
Comment (kilonzi) [inline pipelines/cohort_screening/src/variants_report.py:413]: Is this supposed to be hardcoded? or passed as argument

File: pipelines/cohort_screening/src/variants_report.py [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+    # remove any positions that don't have any samples left
+    for position in filtered_positions[:]:
+        if 'samples' in position and len(position['samples']) == 0:
+            logging.info(f'position has no samples left, removing position' + str(position))
+            filtered_positions.remove(position)
+            removed_positions += 1
+
+    # # for testing
+    # with open("filtered_positions_criteria_1.json", 'w') as outfile:
+    #     json.dump(filtered_positions, outfile, indent=4)
```
Comment (kilonzi) [inline pipelines/cohort_screening/src/variants_report.py:202]: Maybe we can move this to _test_variants_report.py_

File: pipelines/cohort_screening/src/variants_report.py [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+    else:
+        with open("../local_testing/filtered_positions_final_test.json", 'w') as outfile:
+            json.dump(filtered_positions, outfile)
+            # json.dump(filtered_positions, outfile, indent=4)  # add indent for pretty print, remove for py reading
+
+    format_report(
+        map_variants_to_table(
+            filter_variants_for_report(filtered_positions)
+        )
+    )
```
Comment (kilonzi) [inline pipelines/cohort_screening/src/variants_report.py:421]: I am sure this works as expected but calling each one on it's own might be good for readability and traceability.

File: pipelines/cohort_screening/src/requirements.txt [hunk 6]
```diff
@@ -0,0 +1,5 @@
+gtfparse
+jinja2
+pandas
```
Comment (kilonzi) [inline pipelines/cohort_screening/src/requirements.txt:3]: Do we want to version these requirements?

File: pipelines/cohort_screening/src/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+COPY . /src
+
+# builds with GitHub Action "Main Validation and Release" ../.github/workflows/build-and-publish-main.yaml
+# tag = us-east4-docker.pkg.dev/broad-gotc-prod/variantreport:${{steps.get-artifact-slug.outputs.slug}}, us-east4-docker.pkg.dev/broad-gotc-prod/variantreport:latest
+
+# For local Dev
+# Google Artifact Registry
+# docker build -t us-east4-docker.pkg.dev/broad-gotc-prod/variantreport:<new_version> .
+# then push, if you want to push
+# Azure Container Registry
```
Comment (kilonzi) [inline pipelines/cohort_screening/src/Dockerfile:25]: Do we want to move this to DEVELOPMENT.md

File: pipelines/cohort_screening/src/MANE.GRCh38.v1.3.ensembl_genomic.gtf.gz
Comment (bahill) [reply pipelines/cohort_screening/src/MANE.GRCh38.v1.3.ensembl_genomic.gtf.gz:1]: I added a comment though - to make this clear

File: pipelines/cohort_screening/src/variants_report.py (hunk 3 above)
Comment (bahill) [reply pipelines/cohort_screening/src/variants_report.py]: AH good catch!

File: pipelines/cohort_screening/src/variants_report.py (hunk 4 above)
Comment (bahill) [reply pipelines/cohort_screening/src/variants_report.py]: I just removed it - if we get to a point where we need to do more heavy dev again we can add it back locally

File: pipelines/cohort_screening/src/requirements.txt (hunk 6 above)
Comment (bahill) [reply pipelines/cohort_screening/src/requirements.txt]: done

File: pipelines/cohort_screening/src/Dockerfile (hunk 7 above)
Comment (bahill) [reply pipelines/cohort_screening/src/Dockerfile:25]: I like having this in the Dockerfile, but I don't feel super strong about that. We need to have a larger conversation with the whole M42 pipelines team about how we want to organize our repositories, so I'll leave this till then for consensus.

**Review by nikellepetrillo — APPROVED** [review]
looks good to me!

**Pull request merged**